keep/providers/pagerduty_provider/pagerduty_provider.py

```diff
@@ -386,6 +527,167 @@ def _notify(
+        service_metadata = {}
+        for service in all_services:
+            service_metadata[service["id"]] = service
+
+        for business_service in all_business_services:
+            service_metadata[business_service["id"]] = business_service
+
+        try:
+            service_map_response = requests.get(
+                url=f"{self.BASE_API_URL}/service_dependencies",
+                headers=self.__get_headers(),
+            )
+            service_map_response.raise_for_status()
+            service_map_response = service_map_response.json()
+        except Exception:
+            self.logger.exception("Error while getting service dependencies")
+            raise
+
+        service_topology = {}
+
+        for relationship in service_map_response.get("relationships", []):
+            # Extract dependent and supporting service details
+            dependent = relationship["dependent_service"]
+            supporting = relationship["supporting_service"]
+
+            if dependent["id"] not in service_topology:
+                service_topology[dependent["id"]] = TopologyServiceInDto(
+                    source_provider_id=self.provider_id,
+                    service=dependent["id"],
+                    display_name=service_metadata[dependent["id"]]["name"],
+                    description=service_metadata[dependent["id"]]["description"],
+                    team=", ".join(
+                        team["name"]
+                        for team in service_metadata[dependent["id"]].get("teams", [])
+                    ),
+                )
+            if supporting["id"] not in service_topology:
+                service_topology[supporting["id"]] = TopologyServiceInDto(
+                    source_provider_id=self.provider_id,
+                    service=supporting["id"],
+                    display_name=service_metadata[supporting["id"]]["name"],
+                    description=service_metadata[supporting["id"]]["description"],
+                    team=", ".join(
+                        team["name"]
+                        for team in service_metadata[supporting["id"]].get("teams", [])
+                    ),
+                )
+            service_topology[dependent["id"]].dependencies[supporting["id"]] = "unknown"
+        return list(service_topology.values())
+
+    def _get_incidents(self) -> list[IncidentDto]:
+        incidents = self.__get_all_incidents_or_alerts()
+        incidents = [
+            self._format_incident({"event": {"data": incident}})
+            for incident in incidents
+        ]
+        return incidents
+
+    @staticmethod
+    def _format_incident(
+        event: dict, provider_instance: "BaseProvider" = None
+    ) -> IncidentDto | list[IncidentDto]:
+
+        # Creating an uuid from incident id.
+        m = hashlib.md5()
+        event = event["event"]["data"]
+        seed = event["id"]
+        m.update(seed.encode("utf-8"))
+        incident_id = uuid.UUID(m.hexdigest())
```

Comment:
One things that is very problematic here and needs some thinking is that we don't have any `original_incident_id` attribute, that is unique per incident (kinda like "fingerprint" for alerts). Every time we will call this function, we will bring the same incidents over and over again.

cc: @shahargl @VladimirFilonov - need your eyes here